src/foam/u2/crunch/CapabilityStore.js

```diff
@@ -0,0 +1,235 @@
+                  .add(self.renderSection(category))
+                  ;
+                this.add(e);
+              }
+            });
+          })
+        .end();
+    },
+    function renderFeatured() {
+      var self = this;
+      return self.E()
+        // Featured Capabilities
+        .add(self.slot(function (featuredCapabilities) {
+          var spot = this.E('span');
+          featuredCapabilities.select().then((result) => {
+            let arr = result.array;
+            let grid = self.E();
+            grid
+              .addClass(self.myClass('four-column-grid'));
+            for ( let i=0 ; i < arr.length ; i++ ) {
+              let cap = arr[i];
+              grid = grid
+                .start('div', { columns: 3 })
+                  .style({
+                    flex: 0,
+                    display: 'flex',
+                    height: 'auto'
+                  })
+                  .start(self.CapabilityFeatureView, { data: cap })
+                    .style({
+                      flex: 0,
+                      height: 'auto'
+                    })
+                  .end()
+                  .on('click', () => {
+                    self.crunchController.launchWizard(cap.id);
+                  })
+                .end()
+                ;
+            }
+            spot.add(grid);
+          });
+          return spot;
+        }))
+        // Capability Store Section Previews
+        .select(self.visibleCategoryDAO, function(category) {
+          var sectionElement = this.E('span');
+          var returnElement = this.E()
+            .start('h3')
+              .add(category.name)
+            .end()
+            .add(sectionElement)
+            ;
+          var previewIdsPromise = self.getCategoryDAO_(category.id).limit(6)
+            .select().then(arraySink => arraySink.array.map(x => x.targetId) );
+
+          previewIdsPromise.then(capabilityIds => {
+            self.visibleCapabilityDAO.where(
+              self.IN(self.Capability.ID, capabilityIds)
+            ).select().then((result) => {
+              let arr = result.array;
+              let grid = self.Grid.create();
+              for ( let i=0 ; i < arr.length ; i++ ) {
+                let cap = arr[i];
+                grid = grid
+                  .start(self.GUnit, { columns: 4 })
+                    .tag(self.CapabilityCardView, { data: cap })
+                    .on('click', () => {
+                      self.crunchController.launchWizard(cap.id);
+                    })
+                  .end()
+                  ;
+              }
+              sectionElement.add(grid);
+            });
+          });
+          return returnElement;
+        })
+        ;
+    },
+    function renderSection(category) {
+      var self = this;
+      var sectionElement = this.E();
+
+      // Promise 'p' reports capability IDs that are in this category
+      var p = self.getCategoryDAO_(category.id)
```

Comment:
Use a projection, which will be simpler and faster.

Reply:
https://nanopay.atlassian.net/browse/NP-1373